### PR TITLE
feature/notifications: add pusher, implement like review notification

### PR DESCRIPTION
**Pull request by K0sen**

### CODE REVIEW

File: frontend/config/dev.env.js [hunk 1]
```diff
@@ -3,7 +3,10 @@ const merge = require('webpack-merge')
 const prodEnv = require('./prod.env')
 
 module.exports = merge(prodEnv, {
-  NODE_ENV: '"development"',
-  MAPBOX_TOKEN: '"'+process.env.MAPBOX_TOKEN+'"',
-  MAPBOX_STYLE: '"'+process.env.MAPBOX_STYLE+'"'
-})
\ No newline at end of file
+    NODE_ENV: '"development"',
```
Comment (nemoipaha) [inline frontend/config/dev.env.js]: no need for changes here
merge(prodEnv, {}), config will be copied from .prod.env

File: frontend/config/prod.env.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 'use strict'
 module.exports = {
-  NODE_ENV: '"production"',
-  MAPBOX_TOKEN: '"'+process.env.MAPBOX_TOKEN+'"',
-  MAPBOX_STYLE: '"'+process.env.MAPBOX_STYLE+'"',
-  SENTRY_LARAVEL_DSN: '"'+process.env.SENTRY_LARAVEL_DSN+'"',
-  PUSHER_APP_KEY: '"'+process.env.PUSHER_APP_KEY+'"',
-  PUSHER_APP_CLUSTER: '"'+process.env.PUSHER_APP_CLUSTER+'"',
-  APP_EVENTS_NAMESPACE: '"'+process.env.APP_EVENTS_NAMESPACE+'"',
+    NODE_ENV: '"production"',
```
Comment (nemoipaha) [inline frontend/config/prod.env.js]: indentation for 2 spaces here is also ok

File: frontend/src/components/navbar/TopNavbar.vue [hunk 3]
```diff
@@ -101,27 +127,61 @@
 </template>
 
 <script>
-import { mapActions, mapGetters } from 'vuex';
+import { mapActions, mapGetters, mapMutations } from 'vuex';
 import NavbarSearchPanel from './NavbarSearchPanel';
 import LanguageSelector from './LanguageSelector';
+import LikeReviewNotification from '@/components/notifications/LikeReviewNotification';
```
Comment (nemoipaha) [inline frontend/src/components/navbar/TopNavbar.vue]: LikeReviewNotification is duplicated 3 times ?

File: frontend/src/components/navbar/TopNavbar.vue [hunk 4]
```diff
@@ -130,16 +190,93 @@ export default {
...
+        toggleNotifications() {
+            this.isNewNotifications = false;
+            this.notificationsDisplay = !this.notificationsDisplay;
+        },
+        hideNotifications() {
+            this.notificationsDisplay = false;
+        },
+        notificationComponent: function(notification) {
+            switch (notification.type) {
+                case 'Hedonist\\Notifications\\LikeReviewNotification':
```
Comment (nemoipaha) [inline frontend/src/components/navbar/TopNavbar.vue]: `Hedonist\\Notifications\\LikeReviewNotification` must be const values in notificationService

File: frontend/src/services/common/websocketService.js [hunk 5]
```diff
@@ -30,3 +30,4 @@ export const updateWsAuthToken = (token) => {
     Echo.connector.pusher.config.auth.headers.Authorization = 'Bearer ' + token;
 };
 
+Pusher.log = (m) => console.log(m);
```
Comment (nemoipaha) [inline frontend/src/services/common/websocketService.js]: necessary?

File: app/Actions/Review/CreateReviewAction.php [hunk 6]
```diff
@@ -46,6 +49,8 @@ public function execute(CreateReviewRequest $request): CreateReviewResponse
         );
 
         broadcast(new ReviewAddEvent($review))->toOthers();
+        $this->userRepository->getById($place->creator_id)
```
Comment (nemoipaha) [inline app/Actions/Review/CreateReviewAction.php]: better notation is
$user = $this->userRepository->getById($place->creator_id);
$user->notify(new ReviewPlaceNotification($place, Auth::user()));

File: database/migrations/2018_09_06_060048_add_notifications_receive_to_user_info.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('user_info', function (Blueprint $table) {
+            $table->dropColumn('notifications');
```
Comment (nemoipaha) [inline database/migrations/2018_09_06_060048_add_notifications_receive_to_user_info.php]: notifications_receive column

File: frontend/src/components/navbar/TopNavbar.vue (hunk 3 above)
Comment (K0sen) [reply frontend/src/components/navbar/TopNavbar.vue]: now is not

File: app/Actions/Place/AddPlace/AddPlaceAction.php [hunk 8]
```diff
@@ -136,4 +139,13 @@ public function savePhotos(array $photos, int $placeId, int $creatorId): void
             ]));
         }
     }
+
+    public function sentNotificationToFollowers(Place $place)
```
Comment (nemoipaha) [inline app/Actions/Place/AddPlace/AddPlaceAction.php]: private method

File: app/Actions/Place/AddPlace/AddPlaceAction.php [hunk 9]
```diff
@@ -136,4 +139,13 @@ public function savePhotos(array $photos, int $placeId, int $creatorId): void
...
+    {
+        foreach ($this->userRepository->getFollowers(Auth::user()) as $user) {
+            if ((bool) $user->info->notifications_receive === true) {
```
Comment (nemoipaha) [inline app/Actions/Place/AddPlace/AddPlaceAction.php]: $user->info is eager loaded?

File: app/Actions/Review/CreateReviewAction.php [hunk 10]
```diff
@@ -49,9 +51,21 @@ public function execute(CreateReviewRequest $request): CreateReviewResponse
...
+        $this->sentNotificationToFollowers($place);
+        $notifiableUser = $this->userRepository->getById($place->creator_id);
+        if ((bool) $notifiableUser->info->notifications_receive === true) {
+            $notifiableUser->notify(new ReviewPlaceNotification($place, Auth::user()));
+        }
 
         return new CreateReviewResponse($review);
     }
+
+    public function sentNotificationToFollowers(Place $place)
```
Comment (nemoipaha) [inline app/Actions/Review/CreateReviewAction.php]: private

File: frontend/src/components/navbar/TopNavbar.vue [hunk 11]
```diff
@@ -68,7 +68,7 @@
                                     </li>
                                 </ul>
                                 <div v-else class="notifications__none">
-                                    No notifications for you
+                                    You don't have new notifications
```
Comment (nemoipaha) [inline frontend/src/components/navbar/TopNavbar.vue]: localized string

File: frontend/src/components/notifications/FollowedUserAddPlaceNotification.vue [hunk 12]
```diff
@@ -0,0 +1,44 @@
+<template>
+    <div class="notification-message">
+        <span class="user">
+            <img class="user__avatar" :src="getUserAvatar">
+            <span class="user__name">
+                <a href="#">{{ getUserName }}</a>
+            </span>
+        </span>
+        <span class="text">added a new</span>
```
Comment (nemoipaha) [inline frontend/src/components/notifications/FollowedUserAddPlaceNotification.vue]: localized

File: app/Actions/Place/AddPlace/AddPlaceAction.php (hunk 9 above)
Comment (K0sen) [reply app/Actions/Place/AddPlace/AddPlaceAction.php]: no

**Pull request merged**